plugins/modules/cloud/scaleway/scaleway_container_registry.py

```diff
@@ -0,0 +1,267 @@
+short_description: Scaleway Container registry management module
+version_added: 5.8.0
+author: Guillaume MARTINEZ (@Lunik)
+description:
+  - This module manages container registries on Scaleway account.
+extends_documentation_fragment:
+  - community.general.scaleway
+  - community.general.scaleway_waitable_resource
+
+
+options:
+  state:
+    type: str
+    description:
+      - Indicate desired state of the container regitry.
+    default: present
+    choices:
+      - present
+      - absent
+
+  project_id:
+    type: str
+    description:
+      - Project identifier.
+    required: true
+
+  region:
+    type: str
+    description:
+      - Scaleway region to use (for example C(fr-par)).
+    required: true
+    choices:
+      - fr-par
+      - nl-ams
+      - pl-waw
+
+  name:
+    type: str
+    description:
+      - Name of the container registry.
+    required: true
+
+  description:
+    description:
+      - Description of the container registry.
+    type: str
+
+  privacy_policy:
+    type: str
+    description:
+      - Default visibility policy.
+      - Everyone will be able to pull images from a C(public) registry.
+    choices:
+      - public
+      - private
+    default: private
+'''
+
+EXAMPLES = '''
+- name: Create a container registry
+  community.general.scaleway_container_registry:
+    project_id: '{{ scw_project }}'
+    state: present
+    region: fr-par
+    name: my-awesome-container-registry
+  register: container_registry_creation_task
+
+- name: Make sure container regitry is deleted
+  community.general.scaleway_container_registry:
+    project_id: '{{ scw_project }}'
+    state: absent
+    region: fr-par
+    name: my-awesome-container-registry
+'''
+
+RETURN = '''
+container_registry:
+    description: The container registry informations.
+    returned: when I(state=present)
+    type: dict
+    sample: {
+      "created_at": "2022-10-14T09:51:07.949716Z",
+      "description": "Managed by Ansible",
+      "endpoint": "rg.fr-par.scw.cloud/my-awesome-registry",
+      "id": "0d7d5270-7864-49c2-920b-9fd6731f3589",
+      "image_count": 0,
+      "is_public": false,
+      "name": "my-awesome-registry",
+      "organization_id": "10697b59-5c34-4d24-8d15-9ff2d3b89f58",
+      "project_id": "3da4f0b2-06be-4773-8ec4-5dfa435381be",
+      "region": "fr-par",
+      "size": 0,
+      "status": "ready",
+      "status_message": "",
+      "updated_at": "2022-10-14T09:51:07.949716Z"
+    }
+'''
+
+from ansible_collections.community.general.plugins.module_utils.scaleway import (
+    SCALEWAY_ENDPOINT, SCALEWAY_REGIONS, scaleway_argument_spec, Scaleway,
+    wait_to_complete_state_transition, scaleway_waitable_resource_argument_spec,
+    resource_attributes_should_be_changed, fetch_all_resources
+)
+from ansible.module_utils.basic import AnsibleModule
+
+STABLE_STATES = (
+    "ready",
+    "absent"
+)
+
+MUTABLE_ATTRIBUTES = (
+    "description",
+    "is_public"
+)
+
+
+def payload_from_wished_cr(wished_cr):
```

Comment:
what is a "cr" in this context?

Reply:
I would prefer a name that better conveys the idea, but I won't push for it. On a quick glance I _think_ something like `build_registry_payload()`. Or maybe `prepare_`. This function is basically converting one dict to another, to satisfy the API.

Reply:
Another random and generic thought around naming in this "Ansible module" context: given that in this file things are not meant to be imported elsewhere, I see no need and little to gain by qualifying the very topic of the module in the things' names. I mean, this is a module that deals with Container Registries. Do we really need to name the payload as "container registry" (or "cr" for short)? Do we have other payloads we are dealing with here, so we need to differentiate?

I understand the "explicit is better than implicit" but IMO even that tenet, if taken to extremes, does more harm than good, in terms of readability.